### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by ishimwepati**
Hi ! 

In this branch we completed the following project requirements:

1. We created an API endpoint to retrieve a list of all posts for a user.
2. We developed an API endpoint to fetch all comments for a user's post.
3. We added a new API endpoint was implemented to allow users to add comments to a post. The comment's owner is automatically set as the user making the comment without the need for an additional parameter.
4. Our API endpoints were configured to receive JSON input and respond with JSON output.
5. JSON Web Tokens (JWT) were integrated to generate tokens for secure authentication.
6. We made necessary updates to the project's readme documentation.
7.  We Identified linter issues were fixed to ensure code quality and compliance with coding standards.

Your suggestions are most welcome!

### CODE REVIEW

File: app/controllers/api/v1/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @current_user = User.where(email: session[:user]['email']).first
+    @post = User.find(params[:user_id]).posts.find(params[:post_id])
+    @comment = @post.comments.new(text: comment_params[:text], user: @current_user)
+
+    if @comment.save
+      render json: { success: true, data: { comment: @comment } }, status: :created
+    else
+      render json: { success: false, errors: @comment.errors.full_messages }, status: :unprocessable_entity
+    end
+  end
```
Comment (mckent05) [inline app/controllers/api/v1/comments_controller.rb]: - You have done well so far, but I have this error when trying to create a comment under a particular post. Please fix this 🙏 
wefwefefwef
- According to the project requirements, the owner of the comment should be the user creating the comment. Please ensure you have properly configured your authentication for your api-endpoints. Good job using the `jwt` gem, but, optionally, I advise you consider looking into using [devise-jwt](https://enlear.academy/how-to-create-a-rails-6-api-with-devise-jwt-46fa35085e85). Please fix this 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @ishimwepati, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job creating api-endpoints for your application 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/v1/comments_controller.rb (hunk 1 above)
Comment (ishimwepati) [reply app/controllers/api/v1/comments_controller.rb]: Thanks, I have seen the error, the actual code I pushed them on an other Branch, 
But now everything is working fine !

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @ishimwepati & @coker2019, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Has a proper Pull Request. ✔️
- Blog app: add API endpoints ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**